eth/downloader/downloader.go

```diff
@@ -1015,11 +1015,16 @@ func (d *Downloader) fetchHeaders(p *peerConnection, from uint64, head uint64) e
 
 	// Start pulling the header chain skeleton until all is done
 	var (
-		skeleton = true  // Skeleton assembly phase or finishing up
-		pivoting = false // Whether the next request is pivot verification
-		ancestor = from
-		mode     = d.getMode()
+		skeleton           = true  // Skeleton assembly phase or finishing up
+		pivoting           = false // Whether the next request is pivot verification
+		ancestor           = from
+		mode               = d.getMode()
+		headerCheckTimer   = time.NewTimer(fsHeaderContCheck)
+		headerDelayedTimer = time.NewTimer(fsHeaderContCheck)
 	)
+	defer headerCheckTimer.Stop()
+	defer headerDelayedTimer.Stop()
```

Comment:
We can't read the timers after this function returns so we don't need to close them (?).

Reply:
> We can't read the timers after this function returns so we don't need to close them (?).

When ```select``` waiting both time.After and cancelCh, the AfterTimer is not recovered by the garbage collector until the timer fires, maybe delay recycled for ```fsHeaderContCheck``` seconds. So using ```NewTimer()``` and ```Stop()```to stop manually when the func() returned is a little efficiency I think.